aries-site/src/examples/templates/filtering/FilteringWithSelect.js

```diff
@@ -0,0 +1,371 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import {
+  Anchor,
+  Box,
+  Button,
+  Layer,
+  List,
+  Header,
+  Heading,
+  Menu,
+  ResponsiveContext,
+  Select,
+  Text,
+  TextInput,
+} from 'grommet';
+import { Filter, FormClose, More, Search } from 'grommet-icons';
+
+const allData = [
+  { name: 'Apex-Server', status: 'Ready' },
+  { name: 'AuroraEdge-7.7.-edgedeploy4', status: 'Paused' },
+  { name: 'ECAAS3', status: 'Ready' },
+  { name: 'ECP-Master-01', status: 'Ready' },
+  { name: 'ECaas-EdgeSite1', status: 'Paused' },
+  { name: 'IronMan-2', status: 'Ready' },
+  { name: 'california_1', status: 'Paused' },
+];
+
+const defaultSelectValue = 'All';
+const defaultFilters = {};
+
+const StyledButton = styled(Button)`
+  border: 1px solid
+    ${({ theme }) => theme.global.colors.border[theme.dark ? 'dark' : 'light']};
+  &:hover {
+    background: transparent;
+  }
+`;
+
+const StyledTextInput = styled(TextInput).attrs(() => ({
+  'aria-labelledby': 'search-icon',
+}))``;
+
+export const FilteringWithSelect = () => {
+  const [data, setData] = useState(allData);
+  const [filtering, setFiltering] = useState(false);
+  const [filters, setFilters] = useState(defaultFilters);
+  const [searchFocused, setSearchFocused] = useState(false);
+  const [search, setSearch] = useState();
+  const inputRef = useRef();
+  const size = useContext(ResponsiveContext);
+
+  useEffect(() => {
+    if (searchFocused && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [searchFocused, setSearchFocused]);
+
+  const filterData = (array, criteria, searchValue = search) => {
+    if (Object.keys(criteria).length) setFiltering(true);
+    else setFiltering(false);
+    setFilters(criteria);
+
+    let filterResults;
+    const filterKeys = Object.keys(criteria);
+    filterResults = array.filter(item => {
+      // validates all filter criteria
+      return filterKeys.every(key => {
+        // ignores non-function predicates
+        if (typeof criteria[key] !== 'function') return true;
+        return criteria[key](item[key]);
+      });
+    });
+
+    if (searchValue) {
+      filterResults = filterResults.filter(o =>
+        Object.keys(o).some(
+          k =>
+            typeof o[k] === 'string' &&
+            o[k].toLowerCase().includes(searchValue.toLowerCase()),
+        ),
+      );
+    }
+    return filterResults;
+  };
+
+  useEffect(() => {
+    if (filters !== defaultFilters) {
+      setFiltering(true);
+    } else {
+      setFiltering(false);
+    }
+  }, [filters, setFiltering]);
+
+  return (
+    <Box
+      gap="medium"
+      width={{ max: 'xxlarge' }}
+      pad={{ horizontal: 'xsmall' }}
+      margin="auto"
+      overflow="auto"
```

Comment:
Move overflow to `<Results />`?

Reply:
Without overflow="auto", the example will stretch outside of the Example bounds.

Reply:
Perhaps wrapping the `<List />` in `<Results />` with a `<Box fill overflow="auto">` similar to what is being done in FilterLists.js?

Reply:
Basically was thinking that if the results overflow, they would be scrollable while the filter controls would still be in view. Similar to the list implementations.